src/main/java/jenkins/plugins/logstash/persistence/LogzioHttpsClient.java

```diff
@@ -0,0 +1,100 @@
+package jenkins.plugins.logstash.persistence;
+
+import io.logz.sender.FormattedLogMessage;
+import io.logz.sender.HttpsRequestConfiguration;
+import io.logz.sender.HttpsSyncSender;
+import io.logz.sender.SenderStatusReporter;
+import io.logz.sender.exceptions.LogzioParameterErrorException;
+import io.logz.sender.exceptions.LogzioServerErrorException;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class LogzioHttpsClient{
+    private static final int MAX_SIZE_IN_BYTES = 8 * 1024 * 1024;  // 8 MB
+    private static final int CONNECT_TIMEOUT = 10 * 1000;
+    private static final int SOCKET_TIMEOUT = 10 * 1000;
+    private final HttpsSyncSender logzioClient;
+    private List<FormattedLogMessage> messages; //Not thread safe
+    private int size;
+
+    LogzioHttpsClient(String token, String listener, String type) throws LogzioParameterErrorException {
+        HttpsRequestConfiguration gzipHttpsRequestConfiguration = HttpsRequestConfiguration
+                .builder()
+                .setLogzioToken(token)
+                .setLogzioType(type)
+                .setLogzioListenerUrl(listener)
+                .setSocketTimeout(SOCKET_TIMEOUT)
+                .setConnectTimeout(CONNECT_TIMEOUT)
+                .setCompressRequests(true)
+                .build();
+
+        logzioClient = new HttpsSyncSender(gzipHttpsRequestConfiguration, new Reporter());
+        messages = new ArrayList<>();
```

Comment:
If not synchronized this will become a problem as soon as you have multiple parallel builds. Use
`Collections.synchronizedList(new ArrayList<>())`
Makes it thread safe probably.